### PR TITLE
set the mlModel info in the deploy stage

### PR DESCRIPTION
**Pull request by Zhangxunmt**
### Description
Set the mlModel info in the deployment. Also use the cached value outside of mlModel to obtain the ModelStatus.
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by zane-neo** [comment]
Is this a critical issue that we need to release in 2.14?

**Review by zane-neo — DISMISSED** [review]

**Comment by zane-neo** [comment]
format code

**Comment by dhrubo-os** [comment]
apply spotless

File: plugin/src/main/java/org/opensearch/ml/model/MLModelCacheHelper.java [hunk 1]
```diff
@@ -386,9 +386,11 @@ public String[] getLocalDeployedModels() {
      */
     public String[] getExpiredModels() {
```
Comment (dhrubo-os) [inline plugin/src/main/java/org/opensearch/ml/model/MLModelCacheHelper.java:387]: Do we have any test for this method?

Comment (Zhangxunmt) [reply plugin/src/main/java/org/opensearch/ml/model/MLModelCacheHelper.java:387]: The syncUpCron will cover this part. Only an unit tests will not find this edge case. It has to be in the scenario of "local model deployed and wait more than 10 seconds to predict".

**Comment by dhrubo-os** [comment]
let's push commit rather than amending the commit.

**Comment by Zhangxunmt** [comment]
> let's push commit rather than amending the commit.

No for spotless. Check the commit history. How many meaningless commits we have for a single PR?

**Review by ylwu-amzn — APPROVED** [review]

**Review by dhrubo-os — APPROVED** [review]
